src/Atmos/Model/AtmosModel.jl

```diff
@@ -1142,4 +1150,250 @@ function numerical_flux_first_order!(
+                    u⁺' * normal_vector + c⁺ - (ũᵀn + c̃_LM),
+                ),
+            ),
+            max(
+                abs(ũᵀn),
+                max(
+                    0,
+                    ũᵀn - (u⁻' * normal_vector),
+                    u⁺' * normal_vector - (ũᵀn),
+                ),
+            ),
+        )
+    end
+    if (numerical_flux.LV == true)
+        #Pseudo LeVeque Fix
+        δ_L_1 = max(0, ũᵀn - ũᵀn⁻)
+        δ_L_2 = max(0, ũᵀn - c̃_LM - (ũᵀn⁻ - c⁻))
+        δ_L_3 = max(0, ũᵀn + c̃_LM - (ũᵀn⁻ + c⁻))
+        δ_R_1 = max(0, ũᵀn⁺ - ũᵀn)
+        δ_R_2 = max(0, ũᵀn⁺ - c⁺ - (ũᵀn - c̃_LM))
+        δ_R_3 = max(0, ũᵀn⁺ + c⁺ - (ũᵀn + c̃_LM))
+        if (ũᵀn < δ_L_1 && ũᵀn > -δ_R_1)
+            qa1 = ((δ_L_1 - δ_R_1) * ũᵀn + 2 * δ_L_1 * δ_R_1) / (δ_L_1 + δ_R_1)
+        else
+            qa1 = abs(ũᵀn)
+        end
+        if (ũᵀn - c̃ < δ_L_2 && ũᵀn - c̃_LM > -δ_R_2)
+            qa2 =
+                ((δ_L_2 - δ_R_2) * (ũᵀn - c̃_LM) + 2 * δ_L_2 * δ_R_2) /
+                (δ_L_2 + δ_R_2)
+        else
+            qa2 = abs(ũᵀn - c̃_LM)
+        end
+        if (ũᵀn + c̃_LM < δ_L_3 && ũᵀn + c̃ > -δ_R_3)
+            qa3 =
+                ((δ_L_3 - δ_R_3) * (ũᵀn + c̃_LM) + 2 * δ_R_3 * δ_R_3) /
+                (δ_L_3 + δ_R_3)
+        else
+            qa3 = abs(ũᵀn + c̃_LM)
+        end
+        Λ = SDiagonal(qa2, qa1, qa1, qa1, qa3, qa1)
+    end
+    if (numerical_flux.LVPP == true)
+        #PosPreserving with LeVeque
+        b_L = min(ũᵀn - c̃_LM, ũᵀn⁻ - c⁻)
+        b_R = max(ũᵀn + c̃_LM, ũᵀn⁺ + c⁺)
+        b⁻ = min(0, b_L)
+        b⁺ = max(0, b_R)
+        δ_L_1 = max(0, ũᵀn - b⁻)
+        δ_L_2 = max(0, ũᵀn - c̃_LM - b⁻)
+        δ_L_3 = max(0, ũᵀn + c̃_LM - b⁻)
+        δ_R_1 = max(0, b⁺ - ũᵀn)
+        δ_R_2 = max(0, b⁺ - (ũᵀn - c̃_LM))
+        δ_R_3 = max(0, b⁺ - (ũᵀn + c̃_LM))
+        if (ũᵀn < δ_L_1 && ũᵀn > -δ_R_1)
+            qa1 = ((δ_L_1 - δ_R_1) * ũᵀn + 2 * δ_L_1 * δ_R_1) / (δ_L_1 + δ_R_1)
+        else
+            qa1 = abs(ũᵀn)
+        end
+        if (ũᵀn - c̃_LM < δ_L_2 && ũᵀn - c̃_LM > -δ_R_2)
+            qa2 =
+                ((δ_L_2 - δ_R_2) * (ũᵀn - c̃) + 2 * δ_L_2 * δ_R_2) /
+                (δ_L_2 + δ_R_2)
+        else
+            qa2 = abs(ũᵀn - c̃_LM)
+        end
+        if (ũᵀn + c̃_LM < δ_L_3 && ũᵀn + c̃_LM > -δ_R_3)
+            qa3 =
+                ((δ_L_3 - δ_R_3) * (ũᵀn + c̃_LM) + 2 * δ_R_3 * δ_R_3) /
+                (δ_L_3 + δ_R_3)
+        else
+            qa3 = abs(ũᵀn + c̃_LM)
+        end
+        Λ = SDiagonal(qa2, qa1, qa1, qa1, qa3, qa1)
+    end
+
+
+
+
+    M = hcat(
+        SVector(1, ũc̃⁺[1], ũc̃⁺[2], ũc̃⁺[3], h̃ - c̃ * ũᵀn, qt),
+        SVector(0, τ1[1], τ1[2], τ1[3], τ1' * ũ, 0),
+        SVector(0, τ2[1], τ2[2], τ2[3], τ2' * ũ, 0),
+        SVector(1, ũ[1], ũ[2], ũ[3], ũ' * ũ / 2 + Φ - _T_0 * _cv_m, 0),
+        SVector(1, ũc̃⁻[1], ũc̃⁻[2], ũc̃⁻[3], h̃ + c̃ * ũᵀn, qt),
+        SVector(0, 0, 0, 0, _e_int_v0, 1),
+    )
```

Comment:
Have you checked M * Λ * M^{-1} = d(F * n)/dU ? i.e., a unit test.  
This is not efficient (compared with the Dry Roe, OK to live with this now)

Reply:
Yes the Schur complement has to be calculated for us to make this as efficient as the DryRoe. We can do this at a later time. Would including the flux in the isentropic vortex test work for this purpose?